gravitee-apim-console-webui/src/organization/configuration/user/detail/org-settings-user-detail.component.spec.ts

```diff
@@ -152,12 +159,37 @@ describe('OrgSettingsUserDetailComponent', () => {
     // No flush to stop test here
   });
 
+  it('should not reset password if no email', async (done) => {
+    const user = fakeUser({
+      id: 'userId',
+      source: 'gravitee',
+      email: null,
+    });
+    expectUserTokensGetRequest(user);
+    expectUserGetRequest(user);
+    expectEnvironmentListRequest();
+    expectUserGroupsGetRequest(user.id);
+    expectUserMembershipGetRequest(user.id, 'api');
+    expectUserMembershipGetRequest(user.id, 'application');
+    expectRolesListRequest('ORGANIZATION');
+
+    const userCard = await loader.getHarness(MatCardHarness.with({ selector: '.org-settings-user-detail__card' }));
+    await userCard.getHarness(MatButtonHarness.with({ text: 'Reset password' })).catch((reason) => {
+      expect(reason.message).toEqual(
+        'Failed to find element matching one of the following queries:\n' +
+          '(MatButtonHarness with host element matching selector: "[mat-button],[mat-raised-button],[mat-flat-button],[mat-icon-button],[mat-stroked-button],[mat-fab],[mat-mini-fab]" satisfying the constraints: text = "Reset password")',
+      );
+      done();
+    });
```

Comment:
Maybe the following is more robust:

```suggestion
    expect(await userCard.getAllHarnesses(MatButtonHarness.with({ text: 'Reset password' }))).toHaveLength(0);
```